src/runtime/pybuffer.cs

```diff
@@ -0,0 +1,228 @@
+        /// <param name="buf">Buffer to copy to</param>
+        /// <returns>0 is returned on success, -1 on error.</returns>
+        public int ToContiguous(IntPtr buf, char order)
+        {
+            if (disposedValue)
+                throw new ObjectDisposedException(nameof(PyBuffer));
+            if (Runtime.pyversionnumber < 36)
+                throw new NotSupportedException("ToContiguous requires at least Python 3.6");
+            return Runtime.PyBuffer_ToContiguous(buf, _handle, _view.len, order);
+        }
+
+        /// <summary>
+        /// Fill the strides array with byte-strides of a contiguous (C-style if order is 'C' or Fortran-style if order is 'F') array of the given shape with the given number of bytes per element.
+        /// </summary>
+        public static void FillContiguousStrides(int ndims, IntPtr shape, IntPtr strides, int itemsize, char order)
+        {
+            Runtime.PyBuffer_FillContiguousStrides(ndims, shape, strides, itemsize, order);
+        }
+
+        /// <summary>
+        /// FillInfo Method
+        /// </summary>
+        /// <remarks>
+        /// Handle buffer requests for an exporter that wants to expose buf of size len with writability set according to readonly. buf is interpreted as a sequence of unsigned bytes.
+        /// The flags argument indicates the request type. This function always fills in view as specified by flags, unless buf has been designated as read-only and PyBUF_WRITABLE is set in flags.
+        /// On success, set view->obj to a new reference to exporter and return 0. Otherwise, raise PyExc_BufferError, set view->obj to NULL and return -1;
+        /// If this function is used as part of a getbufferproc, exporter MUST be set to the exporting object and flags must be passed unmodified.Otherwise, exporter MUST be NULL.
+        /// </remarks>
+        /// <returns>On success, set view->obj to a new reference to exporter and return 0. Otherwise, raise PyExc_BufferError, set view->obj to NULL and return -1;</returns>
+        public int FillInfo(IntPtr exporter, IntPtr buf, long len, int _readonly, int flags)
+        {
+            if (disposedValue)
+                throw new ObjectDisposedException(nameof(PyBuffer));
+            return Runtime.PyBuffer_FillInfo(_handle, exporter, buf, (IntPtr)len, _readonly, flags);
+        }
+
+        /// <summary>
+        /// Writes a managed byte array into the buffer of a python object. This can be used to pass data like images from managed to python.
+        /// </summary>
+        public void Write(byte[] buffer, int offset, int count)
+        {
+            if (disposedValue)
+                throw new ObjectDisposedException(nameof(PyBuffer));
+            if (ReadOnly)
+                throw new InvalidOperationException("Buffer is read-only");
+            if (count > buffer.Length)
+                throw new ArgumentOutOfRangeException("count", "Count is bigger than the buffer.");
```

Comment:
Might also want to check, that `offset` is non-negative, and that `offset + count`  is `<= buffer.Length`